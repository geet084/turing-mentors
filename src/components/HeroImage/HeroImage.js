import React from 'react';
import pic from '../../images/hero-image.png';

export const HeroImage = () => {
  return (
    <header>
      <img className="hero-image" src={pic} alt="desk" />
      <div className="hero-image-bar">
        <div className="color-left"></div>
        <div className="color-mid"></div>
        <div className="color-right"></div>
      </div>
    </header>
  )
}

export default HeroImage;
