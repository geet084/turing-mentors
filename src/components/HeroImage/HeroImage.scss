.hero-image {
  width: 100vw;
}

.hero-image-bar {
  display: flex;
  margin-top: -3px;
}

.hero-image-bar div {
  height: 1.25vh;
  width: 33.33%;
}

.color-left {
  background-color: #f9ae06;
}

.color-mid {
  background-color: #ef3852;
}

.color-right {
  background-color: #12f1fc;
}