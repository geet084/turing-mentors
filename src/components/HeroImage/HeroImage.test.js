import React from 'react';
import { HeroImage } from './HeroImage';
import { shallow } from 'enzyme';

describe('HeroImage', () => {
  let wrapper;
  
  it('should match the correct snapshot', () => {
    wrapper = shallow(<HeroImage />)
    expect(wrapper).toMatchSnapshot()
  });
});