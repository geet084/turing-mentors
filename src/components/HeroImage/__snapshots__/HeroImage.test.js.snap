// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroImage should match the correct snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeroImage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <img
          alt="desk"
          className="hero-image"
          src="hero-image.png"
        />,
        <div
          className="hero-image-bar"
        >
          <div
            className="color-left"
          />
          <div
            className="color-mid"
          />
          <div
            className="color-right"
          />
        </div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "desk",
          "className": "hero-image",
          "src": "hero-image.png",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="color-left"
            />,
            <div
              className="color-mid"
            />,
            <div
              className="color-right"
            />,
          ],
          "className": "hero-image-bar",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "color-left",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "color-mid",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "color-right",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
        ],
        "type": "div",
      },
    ],
    "type": "header",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <img
            alt="desk"
            className="hero-image"
            src="hero-image.png"
          />,
          <div
            className="hero-image-bar"
          >
            <div
              className="color-left"
            />
            <div
              className="color-mid"
            />
            <div
              className="color-right"
            />
          </div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "desk",
            "className": "hero-image",
            "src": "hero-image.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="color-left"
              />,
              <div
                className="color-mid"
              />,
              <div
                className="color-right"
              />,
            ],
            "className": "hero-image-bar",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "color-left",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "color-mid",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "color-right",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
          ],
          "type": "div",
        },
      ],
      "type": "header",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
